src/webapp/pages/form-page/disease-outbreak-event/useDiseaseOutbreakEventForm.ts

```diff
@@ -120,30 +119,40 @@ export function useDiseaseOutbreakEventForm(diseaseOutbreakEventId?: Id): State
         setIsLoading(true);
         setIsSaved(false);
 
-        const diseaseOutbreakEventData = mapFormStateToEntityData(
-            formState.data,
-            currentUser.username,
-            diseaseOutbreakEventWithOptions
-        );
-
-        compositionRoot.diseaseOutbreakEvent.save.execute(diseaseOutbreakEventData).run(
-            () => {
-                setGlobalMessage({
-                    text: i18n.t(`Disease Outbreak saved successfully`),
-                    type: "success",
-                });
-                setIsLoading(false);
-                setIsSaved(true);
-            },
-            err => {
-                setGlobalMessage({
-                    text: i18n.t(`Error saving disease outbreak: ${err.message}`),
-                    type: "error",
-                });
-                setIsLoading(false);
-                setIsSaved(false);
-            }
-        );
+        try {
+            const diseaseOutbreakEventData = mapFormStateToEntityData(
+                formState.data,
+                currentUser.username,
+                diseaseOutbreakEventWithOptions
+            );
+
+            compositionRoot.diseaseOutbreakEvent.save.execute(diseaseOutbreakEventData).run(
+                () => {
+                    setGlobalMessage({
+                        text: i18n.t(`Disease Outbreak saved successfully`),
+                        type: "success",
+                    });
+                    setIsLoading(false);
+                    setIsSaved(true);
+                },
+                err => {
+                    setGlobalMessage({
+                        text: i18n.t(`Error saving disease outbreak: ${err.message}`),
+                        type: "error",
+                    });
+                    setIsLoading(false);
+                    setIsSaved(false);
+                }
+            );
+        } catch (error) {
```

Comment:
Is this defensive programming or we know an scenario to get here? With Future methods, nothing should raise runtime errors.

Reply:
Technically this error is never going to happen because it's the one inside `mapFormStateToEntityData` function when `!dataSource || !incidentStatus`, but the user only can save when the form is valid and in that validation that activates the save button checks that these values are ok.